### PR TITLE
Remove product tests which does not improve coverage

### PR DESCRIPTION
**Pull request by kokosing**
Remove product tests which does not improve coverage

### CODE REVIEW

**Comment by kokosing** [comment]
A lot of product tests does not improve coverage as they are already covered with test on lower layers. 

Generally, it is a bad approach to test corner SQL cases with product tests. Product test should be a last resort tests which should be kind of smoke test that tests all the components work together as a product. Also they should cover things which are not possible to be tested in lower layers, like CLI user experience, 3rd party system integration (like kerberos) etc.

See https://codingjourneyman.com/2014/09/24/the-clean-coder-testing-strategies/, product tests should be at the top of the piramid.

File: presto-product-tests/src/main/resources/sql-tests/testcases/horology_functions/atTimeZoneAccessible.sql [hunk 1]
```diff
@@ -1,3 +1,3 @@
 -- database: presto; groups: qe, horology_functions
 SELECT timezone_hour(TIMESTAMP '2001-08-22 03:04:05.321' at time zone 'Asia/Oral'),
-       timezone_minute(TIMESTAMP '2001-08-22 03:04:05.321' at time zone 'Asia/Oral')
\ No newline at end of file
+       timezone_minute(TIMESTAMP '2001-08-22 03:04:05.321' at time zone 'Asia/Oral')
```
Comment (ArturGajowy) [inline presto-product-tests/src/main/resources/sql-tests/testcases/horology_functions/atTimeZoneAccessible.sql]: Unrelated change

**Comment by kokosing** [comment]
Also there is an initiative to refactor `AbstractTestQueries` so it be easier to improve SQL coverage at that layer, see https://github.com/prestodb/presto/issues/6910

File: presto-product-tests/src/main/resources/sql-tests/testcases/color_functions/bar.sql [hunk 2]
```diff
@@ -1,2 +0,0 @@
--- database: presto; groups: color;
-SELECT bar(1, 10), bar(1, 10, rgb(255, 0, 0), color('#0f0'))
```
Comment (pnowojski) [inline presto-product-tests/src/main/resources/sql-tests/testcases/color_functions/bar.sql]: Is there any sql test for color functions? I see only unit tests `TestColorFunctions`. Maybe those unit tests should be rewritten to actually execute some sql.

**Comment by ArturGajowy** [comment]
I think it'd be easier to review if the single commit was split by categories of tests removed, each with a pointer to the tests whose coverage is duplicated by the removed product tests.

**Comment by losipiuk** [comment]
+1

One question. Did you check if the tests you are removing are actually covered elsewhere. I think we should at lest ma a list of stuff which needs to be added to unit tests as part of this ticket.

**Comment by kokosing** [comment]
Before I apply comments, I would like to hear other think about it. @mattsfuller ?

**Comment by kokosing** [comment]
This is not going to be merged, closing.
